validator/validators/performance/performance_stats.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package performance
+
+import (
+	"log"
+	"math"
+	"sort"
+)
+
+type Stats struct {
+	Average float64
+	P99     float64 //99% percent process
+	Max     float64
+	Min     float64
+	Period  int //in seconds
+	Std     float64
+}
+
+/*
+CalculateMetricStatisticsBasedOnDataAndPeriod takes in an array of data and returns the average, min, max, p99, and stdev of the data.
+statistics are calculated this way instead of using GetMetricStatistics API because GetMetricStatistics would require multiple
+API calls as only one metric can be requested/processed at a time whereas all metrics can be requested in one GetMetricData request.
+*/
+func CalculateMetricStatisticsBasedOnDataAndPeriod(data []float64, dataPeriod float64) Stats {
```

Comment:
It seems a bit silly to have a function like this since CloudWatch will return avg, min, max, p99 for us. No need to change anything now, but maybe there is a way to just query this information from CloudWatch... For example if the performance benchmark runs for an hour (beginning at the 0th minute of a wall clock hour) then we just query CW for the metric(s) with an hourly period.

Better yet, we basically be pulling as much information out of CloudWatch about the performance benchmark and storing it in DynamoDB. That way we could overlay time series graphs from multiple test runs and see how P99 changes over time instead of just a single P99 value for the entire test run.